src/jaegertracing/testutils/MockAgent.cpp

```diff
@@ -31,10 +31,29 @@
 #include "jaegertracing/utils/ErrorUtil.h"
 #include "jaegertracing/utils/UDPClient.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable : 4267)
+#pragma warning(disable : 4244)
+#endif
+
 namespace jaegertracing {
 namespace testutils {
 namespace {
 
+struct SocketUtils {
```

Comment:
What is this doing here again? And this one uses consistent behavior across platforms!

Reply:
I will the remove those utility classes.
* I will replace write by ::send
* as for the read, I will keep two methods just to ajust the return value from signed to unsigned int.

Reply:
OK great. Just check for errors (negative numbers) before casting please.

Reply:
in fact, I used std::max(0, readBytesSize). This allowed t return 0 in case of error. Is that ok for you?

Reply:
I don't understand. How does windows indicate a read was unsuccessful as opposed to successful but zero bytes read.

Reply:
Here is the doc for both
* Linux
  * http://man7.org/linux/man-pages/man2/recvmsg.2.html
  * recv return an unsigned int
  * In case of error, it returns -1 (that will underflow to max int)
* Windows 
  * https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-recv
  * recv return an int
  * In case of error, it returns -1

Now that I am answering you, I find myself thinking that I should test again the error value and return 0 or the value.

Reply:
Here is the doc for both
* Linux
  * http://man7.org/linux/man-pages/man2/recvmsg.2.html
  * recv return an unsigned int
  * In case of error, it returns -1 (that will underflow to max int)
* Windows 
  * https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-recv
  * recv return an int
  * In case of error, it returns -1

Now that I am answering you, I find myself thinking that I should test again the error value and return 0 or the value.

Reply:
Looking through the original code, I see we handled the case of negative values correctly. My suggestion is just to replace `read`/`write` with `recv`/`send` where feasible. The return values will be handled correctly.

Reply:
For the record, `ssize_t` is a *signed* size type, so there will be no underflow when it returns -1.